rust-runtime/aws-smithy-http-server/src/rejection.rs

```diff
@@ -264,21 +264,38 @@ convert_to_request_rejection!(hyper::Error, HttpBody);
 // Required in order to accept Lambda HTTP requests using `Router<lambda_http::Body>`.
 convert_to_request_rejection!(lambda_http::Error, HttpBody);
 
-/// A sum type rejection, implementing [`IntoResponse`] when both variants do.
-pub enum EitherRejection<Left, Right> {
-    Left(Left),
-    Right(Right),
-}
+macro_rules! any_rejection {
+    ($name:ident, $($var:ident),+) => (
+        pub enum $name<$($var),*> {
+            $($var ($var),)*
+        }
 
-impl<P, L, R> IntoResponse<P> for EitherRejection<L, R>
-where
-    L: IntoResponse<P>,
-    R: IntoResponse<P>,
-{
-    fn into_response(self) -> http::Response<crate::body::BoxBody> {
-        match self {
-            EitherRejection::Left(left) => left.into_response(),
-            EitherRejection::Right(right) => right.into_response(),
+        impl<P, $($var,)*> IntoResponse<P> for $name<$($var),*>
+        where
+            $($var: IntoResponse<P>,)*
+        {
+            #[allow(non_snake_case)]
+            fn into_response(self) -> http::Response<crate::body::BoxBody> {
+                match self {
+                    $($name::$var ($var) => $var.into_response(),)*
+                }
+            }
         }
-    }
+    )
+}
+
+pub mod any_rejections {
```

Comment:
Should we `doc(hidden)` this?

Reply:
The [entire parent module is doc(hidden)](https://github.com/awslabs/smithy-rs/blob/06c23f65cc54e69a6cfdc2526e1e491ba77e5dca/rust-runtime/aws-smithy-http-server/src/lib.rs#L24), so it should not be necessary.